Automattic/jetpack
# AI Extension: handle Jetpack Form button instances when composing via AI assistant

This PR checks the number of Button instances when the Jetpack Form is created by using the AI assistant:

* - if it exists twice or more, remove all less the last one
* - if it does not exist, create one.

Fixes https://github.com/Automattic/jetpack/issues/32562

## Proposed changes:

* AI Extension: handle Jetpack Form button instances when composing via AI assistant

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






It is challenging to test because usually, the extension handles the Button block instances correctly. What I did was [commenting this line](https://github.com/Automattic/jetpack/blob/901c2dd2c65893545271d5c713c5fae4db969d31/projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/prompt/index.ts#L261), trying to do not to get the Button in the response.

* Go to the block editor
* Create a Jetpack Form block instance
* Ask for a Form without a Button form
* **After**: The button is added when the response doesn't contain the button. You can check that in the dev console.

https://github.com/Automattic/jetpack/assets/77539/8a6b2e0e-d32f-4b61-9807-f9becd07e7cc

* Ask for a Form with many jetpack button blocks
* Ensure the app corrects the instances:

https://github.com/Automattic/jetpack/assets/77539/ef78381f-a6af-4e95-826f-480f2903a1fd

* Confirm that always, in the end, the Form contains only one Button block instance.
